lang/src/org/partiql/lang/eval/physical/window/NavigationWindowFunction.kt

```diff
@@ -0,0 +1,37 @@
+package org.partiql.lang.eval.physical.window
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.physical.EvaluatorState
+import org.partiql.lang.eval.physical.SetVariableFunc
+import org.partiql.lang.eval.physical.operators.ValueExpression
+
+/**
+ * This abstract class holds some common logic for navigation window function, i.e., LAG, LEAD
+ *
+ * TODO: When we support FIRST_VALUE, etc, we probably need to modify the process row function, since those function requires frame
+ */
+abstract class NavigationWindowFunction() : WindowFunction {
```

Comment:
I'm curious whether NavigationWindowFunction and Lag/Lead need to be public.

Reply:
NavigationWindowFunction may or may not be. The idea is that if a customer want to create their own "custom window function", he or she could just pick an abstract class as the base class and implement their own codes. Kind of similar to our `ExprFunc`. `Lag/Lead` do not need to be public. Good call.